antarest/study/storage/rawstudy/model/filesystem/matrix/input_series_matrix.py

```diff
@@ -105,7 +105,14 @@ def write_dataframe(self, df: pd.DataFrame) -> None:
         if self.default_empty is not None and np.array_equal(df.to_numpy(dtype=np.float64), self.default_empty):
             self.config.path.touch(exist_ok=True)
         else:
-            df.to_csv(self.config.path, sep="\t", header=False, index=False)
+            # We're checking if the dataFrame could be represented with integer values.
+            # If so, we'll write it this way as it would be quicker and the file takes less place on the fs.
+            df_as_int = df.astype(np.int32)
+            try:
+                pd.testing.assert_frame_equal(df, df_as_int, check_dtype=False)
```

Comment:
Maybe we should add `check_exact=True` to make explicit we don't compare with some tolerance (I think it's the case by default, but not that clear).

From what I understand, under the hood we end up converting the columns to numpy arrays ad just performe a `==` on them, so we could implement it this way, but for me it's ok to keep that pandas method.